app/jobs/course/assessment/submission/fetch_submissions_from_koditsu_job.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+class Course::Assessment::Submission::FetchSubmissionsFromKoditsuJob <
+  ApplicationJob
+  include TrackableJob
+  include Rails.application.routes.url_helpers
+  include Course::Assessment::Submission::KoditsuSubmissionsConcern
+
+  protected
+
+  def perform_tracked(assessment_id, updated_at)
+    assessment = Course::Assessment.find_by(id: assessment_id)
+
+    is_koditsu = assessment.is_koditsu_enabled && assessment.koditsu_assessment_id
+    return unless is_koditsu && Time.zone.at(assessment.updated_at) == Time.zone.at(updated_at)
```

Comment:
Can confirm that the time objects don't have any floating point differences?

Reply:
this `updated_at` serves as a token, so that if we update the assessment, it won't fetch submissions over and over.. so yes, using `Time.zone.at` here will not be a problem

Reply:
I know that we're using it to check for job validity. What I'm asking is whether there is any loss of precision after passing the datetimestamp into the job: https://stackoverflow.com/a/29551169  
Have you checked? can confirm no issue?

Reply:
Ah I understand what u ask here.. yes no precision will be lost (all the components regarding datetime, which is Day, Month, Year, HH:MM:SS up to milliseconds and also info regarding UTC is still preserved)

the only thing missing is the timezone (the updated_at assumes it to be UTC+0, and hence the Time.zone.at)